server/ServerFSM.cpp

```diff
@@ -2324,6 +2324,20 @@ sc::result ProcessingTurn::react(const ProcessTurn& u) {
         }
     }
 
+    if (server.IsHostless() && GetOptionsDB().Get<bool>("save.auto.hostless.enabled")) {
+        boost::filesystem::path autosave_dir_path = GetServerSaveDir() / "auto";
+        const auto& extension = MP_SAVE_FILE_EXTENSION;
+        // Add timestamp to autosave generated files
+        std::string datetime_str = FilenameTimestamp();
+
+        std::string save_filename = boost::io::str(boost::format("FreeOrion_%04d_%s%s") % server.CurrentTurn() % datetime_str % extension);
+
+        boost::filesystem::path save_path(autosave_dir_path / save_filename);
+
+        PlayerConnectionPtr dummy_connection = nullptr;
```

Comment:
is this temporary local necessary vs. just passing `nullptr` directly?

Reply:
It's necessary as message constructor accepts non-const reference to connection.

Reply:
How long does it expect that reference to be valid? Beyond the lifetime of `dummy_connection`?

Reply:
It's need to be valid until `MessageEventBase` constructor so it don't have to be valid outside its scope.

Reply:
Should probably change `MessageEventBase` to take a `const PlayerConnectionPtr&` then... doesn't seem to need to be a non-const pointer (but does need to be a pointer-to-non-const).

Reply:
Should it be done in this Pull-Request?

Reply:
I guess it's not necessary.